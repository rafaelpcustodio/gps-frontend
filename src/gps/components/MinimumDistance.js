import PropTypes from 'prop-types'
import React from 'react'
import StyledMinimumDisance from './StyledMinimumDistance'


const MinimumDistance = props => {

    const { distance } = props
    
    return (
        <StyledMinimumDisance>
            { `Closest opened Service distance: ${distance}m` }
        </StyledMinimumDisance>
    )
}

MinimumDistance.propTypes = {
    distance: PropTypes.string
}

export default MinimumDistance