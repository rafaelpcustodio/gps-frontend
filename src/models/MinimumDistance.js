import Location from './Location'

class MinimumDistance {

    constructor({ location, minimumDistance }) {
        this._location = new Location(location)
        this._distance = minimumDistance
    }

    get distance() {
        return this._distance
    }

    get location() {
        return this._location
    }

    get toGoogleMaps() {
        return this.location.toGoogleMaps
    }
}

export default MinimumDistance