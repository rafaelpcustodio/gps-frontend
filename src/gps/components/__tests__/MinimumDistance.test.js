import configureMockStore from 'redux-mock-store'
import { Map, Record } from 'immutable'
import React from 'react'
import renderer from 'react-test-renderer'

import { Provider } from 'react-redux'

import MinimumDistance from '../MinimumDistance'

describe('Tests for Minimum Distance component', () => {
    let store = {}
    let mockStore = {}

    it('Should render without crash', () => {
        mockStore = configureMockStore()
        store = mockStore(Record({
            gpsReducers: Map({
            })
        })())

        const tree = renderer.create(
            <Provider store={ store }>
                <MinimumDistance
                    distance={123}
                />
            </Provider>
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })
})
