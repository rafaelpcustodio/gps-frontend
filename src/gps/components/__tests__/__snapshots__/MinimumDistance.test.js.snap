// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Minimum Distance component Should render without crash 1`] = `
<div
  className="StyledMinimumDistance__StyledMinimumDisance-vqza48-0 kYkUOE"
>
  Closest opened Service: 123m
</div>
`;
